resources/charts/stolon/templates/hooks/create-cluster-job.yaml

```diff
@@ -0,0 +1,89 @@
+            - "-exc"
+            - |
+              export ETCD_ENDPOINTS=${NODE_NAME}:2379
+              /usr/bin/etcd gateway start --endpoints=$ETCD_ENDPOINTS --listen-addr=127.0.0.1:2379 --trusted-ca-file=/var/state/root.cert &
+              while ! etcdctl --endpoints {{ .Values.store.endpoints }} \
+              {{ if not (empty .Values.store.caFile) }}--ca-file={{ .Values.store.caFile }}{{ end }} \
+              {{ if not (empty .Values.store.certFile) }}--cert-file={{ .Values.store.certFile }}{{ end }} \
+              {{ if not (empty .Values.store.key) }}--key-file={{ .Values.store.key }}{{ end }} \
+              cluster-health; do sleep 1 && echo -n .; done
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: etcd-secrets
+              mountPath: /home/stolon/secrets/etcd
+  {{- end }}
+      containers:
+        - name: {{ template "stolon.fullname" . }}-create-cluster
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ["/usr/bin/dumb-init", "--", "/bin/bash", "-exc"]
+          args:
+            - |
+              export ETCD_ENDPOINTS=${NODE_NAME}:2379
+              /usr/bin/etcd gateway start --endpoints=$ETCD_ENDPOINTS --listen-addr=127.0.0.1:2379 --trusted-ca-file=/var/state/root.cert &
+              while ! etcdctl --endpoints {{ .Values.store.endpoints }} \
+              {{ if not (empty .Values.store.caFile) }}--ca-file={{ .Values.store.caFile }}{{ end }} \
+              {{ if not (empty .Values.store.certFile) }}--cert-file={{ .Values.store.certFile }}{{ end }} \
+              {{ if not (empty .Values.store.key) }}--key-file={{ .Values.store.key }}{{ end }} \
+              cluster-health; do sleep 1 && echo -n .; done
+
+              stolonctl init --cluster-name={{ template "stolon.clusterName" . }} \
+              --store-backend={{ .Values.store.backend }} \
+              {{- if eq .Values.store.backend "kubernetes" }}
+              --kube-resource-kind={{ .Values.store.kubeResourceKind }} \
+              {{- else }}
+              --store-endpoints={{ .Values.store.endpoints }} \
+              {{ if not (empty .Values.store.caFile) }}--store-ca-file={{ .Values.store.caFile }}{{ end }} \
+              {{ if not (empty .Values.store.certFile) }}--store-cert-file={{ .Values.store.certFile }}{{ end }} \
+              {{ if not (empty .Values.store.key) }}--store-key={{ .Values.store.key }}{{ end }} \
+              {{- end }}
+              --yes \
+              '{ "initMode": "new", {{- range $key, $value := .Values.clusterSpec }} {{ $key | quote }}: {{ if typeIs "string" $value }} {{ $value | quote }} {{ else if typeIs "bool"  $value }} {{ $value }} {{ else }} {{ int $value }} {{ end }}, {{- end }} "pgParameters": {{ toJson .Values.pgParameters }} }'
```

Comment:
Is this handled by a built-in translator or can it be extended? I mean whether we can add a proper format function instead of embedding a lot of logic into the template with `typeIs` and such?